usdigitalresponse/grants-ingest
# Fix: typos in S3 lifecycle rules

### Ticket #271 

## Description

This PR fixes several typos that are currently preventing non-current versions of S3 objects from being properly transitioned to Glacier and/or expired after an expected duration.

Note: When running Terraform CI for this PR, I ran into GitHub's 65535(?)-character limit. I updated the workflow to truncate plans to 65300 characters to work around the issue, although the solution indicated in #66 would be even better, since unnecessary Lambda builds contribute unnecessarily to plan length.

## Testing

Note: You can inspect the plan output in the [Terraform Summary comment](https://github.com/usdigitalresponse/grants-ingest/pull/272#issuecomment-1684645144) below to verify that the lifecycle rules are being configured correctly.

1. Deploy these terraform changes (e.g. to a localstack environment)
2. Verify that the buckets' lifecycle rules include steps to transition objects to Glacier (where configured in `terraform/main.tf`)
3. Verify that the buckets' lifecycle rules include steps to delete non-current objects (where configured in `terraform/main.tf`)

### Automated and Unit Tests

### Manual tests for Reviewer

## Checklist
